### PR TITLE
Extracting golden-tests aka SQL tests into their own workspace.

### PR DESCRIPTION
**Pull request by sumerman**
## Description

A prerequisite for https://github.com/timescale/promscale_extension/issues/276

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] ~CHANGELOG entry for user-facing changes~
- [x] Updated the relevant documentation

### CODE REVIEW

File: alpine.Dockerfile [hunk 1]
```diff
@@ -65,8 +65,9 @@ USER postgres
 
 # Pre-build extension dependencies
 RUN cd ../ && cargo pgx new promscale && cd promscale
-COPY Cargo.* Makefile /build/promscale/
+COPY Cargo.* Makefile *.sh /build/promscale/
```
Comment (JamesGuthrie) [inline alpine.Dockerfile:68]: Why is this necessary?

File: sql-tests/tests/snapshots/tests__testdata__create_ingest_temp_table.sql.snap [hunk 2]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (JamesGuthrie) [inline sql-tests/tests/snapshots/tests__testdata__create_ingest_temp_table.sql.snap:1]: Why are these files not registered as a "move"? Do we also need to remove them from `e2e`?

Comment (sumerman) [reply sql-tests/tests/snapshots/tests__testdata__create_ingest_temp_table.sql.snap]: Thank you!

File: alpine.Dockerfile (hunk 1 above)
Comment (sumerman) [reply alpine.Dockerfile:68]: Not strictly necessary, but clears warning about missing `extract-extension-version.sh` during docker build

Comment (JamesGuthrie) [reply alpine.Dockerfile:68]: Okay. I would have a _slight_ preference to copy exact file matches (as opposed to globs), so that we know what we're copying in. Being that this is a "build image" (which we copy the compiled artifacts from), I'm not really too concerned.

**Review by JamesGuthrie — APPROVED** [review]

File: ha.Dockerfile [hunk 3]
```diff
@@ -59,9 +59,11 @@ RUN chown -R postgres:postgres /build
 USER postgres
 
 COPY --chown=postgres:postgres Cargo.* /build/promscale/
+COPY --chown=postgres:postgres extract-extension-version.sh /build/promscale/
```
Comment (JamesGuthrie) [inline ha.Dockerfile:62]: This duplicates the line below it.

Comment (sumerman) [reply ha.Dockerfile:62]: dammit! thank you!
